interleave.py

```diff
@@ -1,23 +1,25 @@
-from typing import Iterable
+from typing import Iterable, Any, Generator
 
 
-def interleave(*args: Iterable) -> list:
+def interleave(*args: Iterable) -> Generator[Any, Any, None]:
     """
     :param args: Iterables.
-    :return: List with zipped values.
+    :return:Generator of List with zipped values.
     """
     return split_zipped_list(list(zip(*args)))
 
 
-def split_zipped_list(zipped_list: list[Iterable]) -> list:
+def split_zipped_list(zipped_list: list[Iterable]) -> Generator[Any, Any, None]:
     """
     Accept list of Iterables and extract all values to one list.
     :param zipped_list: List of lists.
-    :return: One lists from all lists.
+    :return:Generator of One lists from all lists.
     """
-    return [item for sub_list in zipped_list for item in sub_list]
+    return (item for sub_list in zipped_list for item in sub_list)
```

Comment:
notice that if it returns a generator - you need to use "yield" instead of "return". So either fix the documentations or the function itself.

Reply:
Hey,
I don't understand.. When I return **( )** I want to return a generator, and to use it I do :
my_generator = interleave(args)
print(next(my_generator ))

Reply:
@YinonTzo the "next" function applies both to generators and iterables. So when you do "next" on this specific function, it in fact activates the "next" of an iterable rather than a generator.
If you want your function to represent a generator, then what you suggested is very good.
Regarding your last question - there is no such this as generator comprehension, as the whole comprehension mechanism is aimed at creating a full list/set/dictionary in advance, which is what a generator is trying to avoid.

Reply:
Thenks for your comment. 
I think now I undersand. 
So to conclusion there is ne somthing like generator stetment ? 
It's really wierd becouse I print the type of my objet and I get 
<generator object split_zipped_list.<locals>.<genexpr> at 0x0000015D0B2D5FC0>

I tried to do somthing like that:
https://www.pythontutorial.net/advanced-python/python-generator-expressions/